src/client/hud.cpp

```diff
@@ -349,34 +345,56 @@ void Hud::drawLuaElements(const v3s16 &camera_offset)
 					inv, e->item, e->dir);
 				break; }
 			case HUD_ELEM_WAYPOINT: {
-				v3f p_pos = player->getPosition() / BS;
```

Comment:
as you rewrote this part, i suggest that you handle it in a proper function for better code reading. This function is quite big and honestly review are very hard.
Thanks

Reply:
fine